src/kzg10/mod.rs

```diff
@@ -80,22 +81,46 @@ impl<E: PairingEngine> KZG10<E> {
         // up to D queries.
         powers_of_gamma_g.push(powers_of_gamma_g.last().unwrap().mul(&beta));
         end_timer!(gamma_g_time);
-        E::G1Projective::batch_normalization(powers_of_g.as_mut_slice());
-        E::G1Projective::batch_normalization(powers_of_gamma_g.as_mut_slice());
+
+        let powers_of_g = E::G1Projective::batch_normalization_into_affine(&powers_of_g);
+        let powers_of_gamma_g = 
+            E::G1Projective::batch_normalization_into_affine(&powers_of_gamma_g);
+
+        // TODO: Add timer for generating negative powers
+        let prepared_neg_powers_of_h =
+            if _produce_g2_powers {
```

Comment:
Because we're actually producing `g2` powers now, we should add change `_produce_g2_powers` to `produce_g2_powers` (remove the underscore at the beginning).